modules/node_groups/locals.tf

```diff
@@ -0,0 +1,11 @@
+locals {
+  # Trying to use `? var.node_groups : {}` does not work
+  node_groups_keys  = toset(var.create_eks ? keys(var.node_groups) : [])
```

Comment:
I thought it was cleaner to have `for_each = local.node_groups_keys` instead of `for_each = toset(var.create_eks ? keys(var.node_groups) : [])` on every resource. 

Cannot directly use `for_each = var.node_groups` as that doesn't support `var.create_eks`.

Cannot use `for_each = var.create_eks ? var.node_groups : {}` because Terraform gets very confused and throws errors depending on what you have in `var.node_groups`. "Object does not equal Object" or the maps are not the same size.

Reply:
I see. Is something like would work ?

```hcl
node_groups_expanded = { for k, v in var.node_groups :  k => merge(
    var.node_groups_defaults,
    v,
  ) if create_eks }
```

I don't test it, but from doc it should generate an empty map if creat_eks is false.

Reply:
🤦‍♂ I see what you mean now. I've made changes.